Library/Homebrew/cask/cmd/zap.rb

```diff
@@ -32,14 +32,7 @@ def self.zap_casks(
         casks.each do |cask|
           odebug "Zapping Cask #{cask}"
 
-          if cask.installed?
-            if (installed_caskfile = cask.installed_caskfile) && installed_caskfile.exist?
-              # Use the same cask file that was used for installation, if possible.
-              cask = CaskLoader.load(installed_caskfile)
-            end
-          else
-            raise CaskNotInstalledError, cask unless force
-          end
+          raise CaskNotInstalledError, cask if !cask.installed? && !force
```

Comment:
Shouldn't we still try to load the installed caskfile and fall back to the one from the tap or the "dummy" cask we've created only if the installed caskfile is unreadable? This would ensure that the correct `zap` stanza is used and not one that's been updated for a newer version of the cask or one that's incomplete. Apologies if I've missed or misunderstood something.

Reply:
@MikeMcQuaid I may also have misunderstood this change, but it seems like this PR may have caused the exact scenario described by @nandahkrishna to occur.

Here's an example from a CI run - https://github.com/Homebrew/homebrew-cask-versions/runs/6654799155?check_suite_focus=true - Namely the uninstall conflicting casks step.
The tapped version of the cask is newer than the installed cask and the uninstall is therefore failing. 
<img width="595" alt="Screen Shot 2022-05-30 at 11 42 40 pm" src="https://user-images.githubusercontent.com/40621599/171004707-c41b0687-529e-497a-9d13-ecb59f729e3d.png">

I have also tested this locally and narrowed it down to the commit associated with the PR.

It can be reproduced by doing the following;
1. `brew install chromedriver`
2. Edit the `chromedriver` cask and bump the version to an arbitrary version
3. `brew uninstall chromedriver` will now fail
```
chromedriver 102.0.5005.61 is still installed.
Remove it with `brew uninstall --cask --force chromedriver`.
```

I have had a quick look, however I cannot find a fix aside from reverting the changes in `cask/cmd/uninstall.rb`

Reply:
@bevanjkay Thanks, good catch. This should be addressed by https://github.com/Homebrew/brew/pull/13348 but please test and let me know.